src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -90,6 +92,28 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             TrackBulkFileOperations();
         }
 
+        private async Task InitializeWorkspaceFailureOutputWindowAsync()
+        {
+            var threadingContext = _componentModel.GetService<IThreadingContext>();
+            var asyncListenerProvider = _componentModel.GetService<IAsynchronousOperationListenerProvider>();
+            var asyncListener = asyncListenerProvider.GetListener(FeatureAttribute.WorkspaceFailureLogger);
+
+            var brokeredServiceContainer = await this.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>().ConfigureAwait(false);
+            Assumes.Present(brokeredServiceContainer);
+            var serviceBroker = brokeredServiceContainer.GetFullAccessServiceBroker();
+            var serviceBrokerClient = new ServiceBrokerClient(serviceBroker, threadingContext.JoinableTaskFactory);
+
+            _workspace.WorkspaceFailed += (sender, eventArgs) =>
+            {
+                var asyncToken = asyncListener.BeginAsyncOperation(nameof(InitializeWorkspaceFailureOutputWindowAsync));
```

Comment:
Isn't WorkspaceFailed already being raised fairly asynchronously? I'm mostly wondering out loud how useful this actually will be...

Reply:
discussed offline - workspace failed event is triggered immediately (generally after file load exception) and not delayed/put into a queue.  So we shouldn't need to wait on the work to be queued before waiting on this.